zkvm/src/vm.rs

```diff
@@ -291,11 +292,16 @@ where
     }
 
     fn neg(&mut self) -> Result<(), VMError> {
-        let mut expr = self.pop_item()?.to_expression()?;
-        for (_, n) in expr.terms.iter_mut() {
-            -*n;
-        }
-        self.push_item(expr);
+        let expr = self.pop_item()?.to_expression()?;
+        let neg_expr = Expression {
+            terms: expr
+                .terms
+                .iter()
+                .map(|t| (t.0, t.1.neg()))
+                .collect::<Vec<_>>(),
+            assignment: expr.assignment,
```

Comment:
Rust newbie sez: does this move ownership of `expr.assignment`? Do I need a different way to do this?

Reply:
my understanding is that yes, the field would take ownership of `expr.assignment`. but I don't see a problem with that - it doesn't seem like `expr.assignment` or `expr` are being used anywhere later in the code, and you already have ownership of `expr`.

Reply:
If you do into_iter, iterator will consume the storage and collect() may reuse it.